app/src/main/java/ch/epfl/skysync/viewmodel/FlightsViewModel.kt

```diff
@@ -0,0 +1,166 @@
+    @Composable
+    fun createViewModel(
+        flightTable: FlightTable,
+        balloonTable: BalloonTable,
+        basketTable: BasketTable,
+        flightTypeTable: FlightTypeTable,
+        vehicleTable: VehicleTable,
+    ): FlightsViewModel {
+      return viewModel<FlightsViewModel>(
+          factory =
+              object : ViewModelProvider.Factory {
+                override fun <T : ViewModel> create(modelClass: Class<T>): T {
+                  return FlightsViewModel(
+                      flightTable,
+                      balloonTable,
+                      basketTable,
+                      flightTypeTable,
+                      vehicleTable,
+                  )
+                      as T
+                }
+              })
+    }
+  }
+
+  private val _currentFlights: MutableStateFlow<List<Flight>> = MutableStateFlow(emptyList())
+  private val _currentBalloons: MutableStateFlow<List<Balloon>> = MutableStateFlow(emptyList())
+  private val _currentBaskets: MutableStateFlow<List<Basket>> = MutableStateFlow(emptyList())
+  private val _currentFlightTypes: MutableStateFlow<List<FlightType>> =
+      MutableStateFlow(emptyList())
+  private val _currentVehicles: MutableStateFlow<List<Vehicle>> = MutableStateFlow(emptyList())
+
+  val currentFlights = _currentFlights.asStateFlow()
+  val currentBalloons = _currentBalloons.asStateFlow()
+  val currentBaskets = _currentBaskets.asStateFlow()
+  val currentFlightTypes = _currentFlightTypes.asStateFlow()
+  val currentVehicles = _currentVehicles.asStateFlow()
+
+  fun refreshAll() {
+    refreshCurrentFlights()
+    refreshCurrentBalloons()
+    refreshCurrentBaskets()
+    refreshCurrentFlightTypes()
+    refreshCurrentVehicles()
+  }
+
+  fun refreshCurrentBalloons() {
+    balloonTable.getAll(
+        { balloons -> _currentBalloons.value = balloons },
+        { exception -> Log.d("Balloonrefresh", exception.toString()) })
+  }
+
+  fun refreshCurrentVehicles() {
+    vehicleTable.getAll(
+        { vehicles -> _currentVehicles.value = vehicles },
+        { exception -> Log.d("Vehiclerefresh", exception.toString()) })
+  }
+
+  fun refreshCurrentBaskets() {
+    basketTable.getAll(
+        { baskets -> _currentBaskets.value = baskets },
+        { exception -> Log.d("Basketrefresh", exception.toString()) })
+  }
+
+  fun refreshCurrentFlightTypes() {
+    flightTypeTable.getAll(
+        { flightTypes -> _currentFlightTypes.value = flightTypes },
+        { exception -> Log.d("FlightTyperefresh", exception.toString()) })
+  }
+
+  fun refreshCurrentFlights() {
+    // todo: check for dirty data (flights added/modified/deleted while offline)
+
+    flightTable.getAll(
+        { flights -> _currentFlights.value = flights },
+        { exception -> Log.d("FLightrefresh", exception.toString()) })
+  }
+
+  /**
+   * modifies the flight by deleting the old flight and adding a new one in the db and the viewmodel
+   */
+  fun modifyFlight(
```

Comment:
There is a function update in FlightTable since the merge of #176 which suits well here